crates/fm/src/lib.rs

```diff
@@ -38,12 +38,21 @@ pub struct FileManager {
 impl FileManager {
     // XXX: Maybe use a AsRef<Path> here, for API ergonomics
     pub fn add_file(&mut self, path_to_file: &Path, file_type: FileType) -> Option<FileId> {
-        let source = file_reader::read_file_to_string(path_to_file).ok()?;
+        // Handle both relative file paths and std/lib virtual paths.
+        let base = Path::new(".").canonicalize().expect("Base path canonicalize failed"); // Should never fail
+        let res = path_to_file.canonicalize().unwrap_or_else(|_| path_to_file.to_path_buf());
+        let resolved_path = res.strip_prefix(base).unwrap_or(&res);
```

Comment:
Retrieving the current directory beforehand with `let base = Path::new(".").canonicalize()`, then calling `canonicalize` on `path_to_file` as well just to strip the base prefix afterward seems a bit roundable.

Can we use absolute paths everywhere instead? E.g. let's use `path_to_file.canonicalize()` without stripping any prefixes.